pftools/python/parflow/tools/io.py

```diff
@@ -453,3 +454,381 @@ def read_clm(file_name, type='clmin'):
+
+    # ---------------------------------------------------------------------------
+    # Region selector
+    # ---------------------------------------------------------------------------
+
+    @property
+    def selector(self):
+        return self._selector
+
+    @selector.setter
+    def set_selector(self, selector):
+        self._selector = selector
+
+    # ---------------------------------------------------------------------------
+    # Grid information
+    # ---------------------------------------------------------------------------
+
+    @property
+    def shape(self):
+        # FIXME do something with selector
+        return (
+            self._run.ComputationalGrid.NZ,
+            self._run.ComputationalGrid.NY,
+            self._run.ComputationalGrid.NX
+        )
+
+    # ---------------------------------------------------------------------------
+    # Slopes X Y Z
+    # ---------------------------------------------------------------------------
+
+    @property
+    def slope_x(self):
+        return self._pfb_to_array(self._run.TopoSlopesX.FileName)
+
+    @property
+    def slope_y(self):
+        return self._pfb_to_array(self._run.TopoSlopesY.FileName)
+
+    @property
+    def slope_z(self):
+        return self._pfb_to_array(self._run.TopoSlopesZ.FileName)
+
+    # ---------------------------------------------------------------------------
+    # Computed Porosity
+    # ---------------------------------------------------------------------------
+
+    @property
+    def computed_porosity(self):
+        return self._pfb_to_array(f'{self._run.get_name()}.out.porosity.pfb')
+
+    # ---------------------------------------------------------------------------
+    # Computed Permeability
+    # ---------------------------------------------------------------------------
+
+    @property
+    def computed_permeability_x(self):
+        return self._pfb_to_array(f'{self._run.get_name()}.out.perm_x.pfb')
+
+    @property
+    def computed_permeability_y(self):
+        return self._pfb_to_array(f'{self._run.get_name()}.out.perm_y.pfb')
+
+    @property
+    def computed_permeability_z(self):
+        return self._pfb_to_array(f'{self._run.get_name()}.out.perm_z.pfb')
+
+    # ---------------------------------------------------------------------------
+    # Pressures
+    # ---------------------------------------------------------------------------
+
+    @property
+    def pressure_initial_condition(self):
+        press_type = self._run.ICPressure.Type
+        if press_type == 'PFBFile':
+            geom_name = self._run.ICPressure.GeomNames
+            if len(geom_name) > 1:
+                raise Exception(f'ICPressure.GeomNames are set to {geom_name}')
+            return self._pfb_to_array(self._run.Geom[geom_name[0]].ICPressure.FileName)
+        else:
+            # HydroStaticPatch, ... ?
+            raise Exception(f'Initial pressure of type {press_type} is not supported')
+
+        return None
```

Comment:
If a function ends with no return statement, it returns `None` by default. So this `return None` is not necessary - but it could be useful for clarity. It's up to you. Also, just plain `return` is the same as `return None` as well.